shacl2md/generator.py

```diff
@@ -0,0 +1,314 @@
+import argparse
+import os
+from itertools import groupby
+from typing import List
+
+from jinja2 import Environment, PackageLoader, select_autoescape
+from lxml import etree
+from plantuml import PlantUML
+from pyshacl import validate
+from rdflib.graph import Graph, URIRef
+from rdflib.namespace import Namespace
+from rdflib.term import Literal
+
+from shacl2md.utilities.lang_labels import get_lang_labels
+from shacl2md.utilities.queries import (CLASS_EXISTS_CHECK, GET_AUTHORS,
+                                        GET_CLASS, GET_CLASSES, GET_DATATYPES,
+                                        GET_DOC_MD, GET_PROPERTIES,
+                                        GET_SUBCLASSES, GET_SUPERCLASSES,
+                                        GET_VALUES)
+from shacl2md.utilities.rdf import RDFClass, to_shortname
+
+SHACL = Namespace("http://www.w3.org/ns/shacl#")
+
+env = Environment(
+    loader=PackageLoader("shacl2md"),
+    autoescape=select_autoescape(),
+    trim_blocks=True,
+)
+
+template = env.get_template("template.md.jinja")
+puml_template = env.get_template("diagram.puml.jinja")
+
+
+
+
+
+class ShaclMarkdownGenerator: 
+
+    def __init__(
+        self,
+        languages : List[str], 
+        output_dir : str= "./", 
+        shacl_shacl_validation : bool = False,
+        version_directory : bool = False, 
+        crosslink_between_graphs : bool = False,
+        jekyll_parent_page : str = "index",
+        jekyll_layout : str = "default",
+        jekyll_nav_order : int = 1,
+        ontology_graphs : List[str] = [],
```

Comment:
Do you expect a graph or a file here? Might be confusing

Reply:
I changed it to accept both Graphs and strings (file location).